src/io/WKTWriter.cpp

```diff
@@ -281,7 +281,7 @@ WKTWriter::appendPointTaggedText(const Point& point, OrdinateSet outputOrdinates
     appendOrdinateText(outputOrdinates, writer);
 
     const CoordinateXY* coord = point.getCoordinate();
-    if (coord == nullptr || (std::isnan(coord->x) && std::isnan(coord->y))) {
```

Comment:
Why not leave this logic in place?  Belt-and-suspenders...

But it should be changed to use `CoordinateXY.isNull`, to centralize this behaviour.  (And maybe that should be changed to use OR rather than AND, to be more inclusive).